File: clients/java_graal/src/main/java/ai_cup_22/model/Loot.java
package ai_cup_22.model;

import ai_cup_22.util.StreamUtil;

/**
 * Loot lying on the ground
 */
public class Loot {
    /**
     * Unique id
     */
    private int id;

    /**
     * Unique id
     */
    public int getId() {
        return id;
    }

    /**
     * Unique id
     */
    public void setId(int value) {
        this.id = value;
    }
    /**
     * Position
     */
    private ai_cup_22.model.Vec2 position;

    /**
     * Position
     */
    public ai_cup_22.model.Vec2 getPosition() {
        return position;
    }

    /**
     * Position
     */
    public void setPosition(ai_cup_22.model.Vec2 value) {
        this.position = value;
    }
    /**
     * Item
     */
    private ai_cup_22.model.Item item;

    /**
     * Item
     */
    public ai_cup_22.model.Item getItem() {
        return item;
    }

    /**
     * Item
     */
    public void setItem(ai_cup_22.model.Item value) {
        this.item = value;
    }

    public Loot(int id, ai_cup_22.model.Vec2 position, ai_cup_22.model.Item item) {
        this.id = id;
        this.position = position;
        this.item = item;
    }

    /**
     * Read Loot from input stream
     */
    public static Loot readFrom(java.io.InputStream stream) throws java.io.IOException {
        int id;
        id = StreamUtil.readInt(stream);
        ai_cup_22.model.Vec2 position;
        position = ai_cup_22.model.Vec2.readFrom(stream);
        ai_cup_22.model.Item item;
        item = ai_cup_22.model.Item.readFrom(stream);
        return new Loot(id, position, item);
    }

    /**
     * Write Loot to output stream
     */
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeInt(stream, id);
        position.writeTo(stream);
        item.writeTo(stream);
    }

    /**
     * Get string representation of Loot
     */
    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder("Loot { ");
        stringBuilder.append("id: ");
        stringBuilder.append(String.valueOf(id));
        stringBuilder.append(", ");
        stringBuilder.append("position: ");
        stringBuilder.append(String.valueOf(position));
        stringBuilder.append(", ");
        stringBuilder.append("item: ");
        stringBuilder.append(String.valueOf(item));
        stringBuilder.append(" }");
        return stringBuilder.toString();
    }
}

File: clients/kotlin_graal/src/main/kotlin/ai_cup_22/model/Loot.kt
package ai_cup_22.model

import ai_cup_22.util.StreamUtil

/**
 * Loot lying on the ground
 */
class Loot {
    /**
     * Unique id
     */
    var id: Int
    /**
     * Position
     */
    var position: ai_cup_22.model.Vec2
    /**
     * Item
     */
    var item: ai_cup_22.model.Item

    constructor(id: Int, position: ai_cup_22.model.Vec2, item: ai_cup_22.model.Item) {
        this.id = id
        this.position = position
        this.item = item
    }

    /**
     * Write Loot to output stream
     */
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, id)
        position.writeTo(stream)
        item.writeTo(stream)
    }

    /**
     * Get string representation of Loot
     */
    override fun toString(): String {
        var stringBuilder = StringBuilder("Loot { ")
        stringBuilder.append("id: ")
        stringBuilder.append(id)
        stringBuilder.append(", ")
        stringBuilder.append("position: ")
        stringBuilder.append(position)
        stringBuilder.append(", ")
        stringBuilder.append("item: ")
        stringBuilder.append(item)
        stringBuilder.append(" }")
        return stringBuilder.toString()
    }

    companion object {
        /**
         * Read Loot from input stream
         */
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): Loot {
            var id: Int
            id = StreamUtil.readInt(stream)
            var position: ai_cup_22.model.Vec2
            position = ai_cup_22.model.Vec2.readFrom(stream)
            var item: ai_cup_22.model.Item
            item = ai_cup_22.model.Item.readFrom(stream)
            return Loot(id, position, item)
        }
    }
}